pkg/api/types.go

```diff
@@ -606,6 +606,8 @@ type HTTPGetAction struct {
 	Port util.IntOrString `json:"port,omitempty"`
 	// Optional: Host name to connect to, defaults to the pod IP.
 	Host string `json:"host,omitempty"`
+	// Optional: Scheme to use for connecting to the host, defaults to HTTP.
+	Scheme string `json:"scheme,omitempty"`
```

Comment:
Unclear whether the intention is that a user put "HTTP" which we understand to mean http:// or whether the user puts "http" which we directly inject into the URL construction?

I think it should be the former - enumerate the schemes we support.  In that case you need a type and enumerated const values for HTTP and HTTPS